testing/fleetservertest/fleetserver_test.go

```diff
@@ -6,21 +6,127 @@ package fleetservertest
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"io"
 	"net/http"
 	"net/url"
-	"strings"
+	"sync"
+	"testing"
+	"time"
 
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/info"
 	"github.com/elastic/elastic-agent/internal/pkg/fleetapi"
 )
 
+// TestRunFleetServer shows how to configure and run a fleet-server capable of
+// enrolling and send actions to a single agent.
+// For a detailed explanation of each piece and how to put them work together,
+// look at ExampleNewServer_checkin_and_ackWithAcker.
+// There is no authentication and the enrollment token is ignored, thus it can
+// be any value.
+func TestRunFleetServer(t *testing.T) {
+	t.Skip("use this test if you want a mock fleet-server running to enroll a real agent")
+	agentID := "agentID"
+	actionID := "ActionID"
+	policyID := "policyID"
+	ackToken := "AckToken"
+	enrolmentToken := "enrolmentToken" //nolint:gosec // it's a test
+	apiKey := APIKey{
+		ID:  "myKeyID",
+		Key: "myvaKeyKey",
+	}
+
+	// FleetHosts needs to be changed after the server is running, so we can
+	// get the port the server is listening on. Therefore, the action generator
+	// captures the 'fleetHosts' variable, so it can read the real fleet-server
+	// address from it.
+	// If you want to predefine an address for the server to listen on, pass
+	// WithAddress(addr) to NewServer.
+	fleetHosts := "host1"
+	var actionsIdx int
+
+	tmpl := TmplPolicy{
+		AckToken: ackToken,
+		AgentID:  agentID,
+		ActionID: actionID,
+		PolicyID: policyID,
+		// FleetHosts needs to be changed after the server is running, so we can
+		// get the port the server is listening on. Therefore, the action generator
+		// captures the 'fleetHosts' variable, so it can read the real fleet-server
+		// address from it.
+		FleetHosts: `"host1", "host2"`,
+		SourceURI:  "http://source.uri",
+		CreatedAt:  "2023-05-31T11:37:50.607Z",
+		Output: struct {
+			APIKey string
+			Hosts  string
+			Type   string
+		}{
+			APIKey: apiKey.String(),
+			Hosts:  `"https://my.clould.elstc.co:443"`,
+			Type:   "elasticsearch"},
+	}
+
+	nextAction := func() (CheckinAction, *HTTPError) {
+		defer func() { actionsIdx++ }()
+		tmpl.FleetHosts = fleetHosts
+
+		actions, err := NewActionPolicyChangeWithFakeComponent(tmpl)
```

Comment:
I think this just returns a single action?

```suggestion
		action, err := NewActionPolicyChangeWithFakeComponent(tmpl)
```